src/panel.js

```diff
@@ -291,6 +326,127 @@ const buildOnboardingPanel = (panelId) => {
   addParagraph(contentWrapper, `${stringId}-p2`);
 
   getLocalizedStrings();
-  page.appendChild(fragment);
+
+  appendFragmentAndSetHeight(page, fragment);
+  addOnboardingListeners(panelId);
+  addDeleteSiteListeners();
+};
+
+
+const addHeaderWithBackArrow = (fragment) => {
+  let el = addHeader(fragment);
+  el = document.createElement("button");
+  el.classList.add("btn-return", "arrow-left");
+  fragment.appendChild(el);
+  return fragment;
+};
+
+
+const defaultAllowedSites = [
+  "instagram.com",
+  "facebook.com",
+  "messenger.com",
+];
+
+
+const makeSiteList = async(fragment, siteList, sitesAllowed=false, addX=false) => {
+
+  // if no sites have been added to the container, show "None"
+  if (siteList.length === 0) {
+    const allowedSiteWrapper = addDiv(fragment, "allowed-site-wrapper");
+    addSpan(allowedSiteWrapper, "no-sites-added");
+    return; 
+  }
+
+  for (const site of siteList) {
+    const allowedSiteWrapper = addDiv(fragment, "allowed-site-wrapper");
+    if (!addX) {
+      allowedSiteWrapper.classList.add("default-allowed-site");
+    }
+
+    let iconDiv = addDiv(allowedSiteWrapper, "allowed-site-icon");
+    if (sitesAllowed) {
+        // should we repeatedly grab these images or download and save them?
+        // need a different place to scoop them up, wondering where activity stream gets their favicons?
```

Comment:
We should download the favicon from the site itself, so we don't make a 3rd-party connection to faviconkit.com from the user's machine.